docs/agent-policy.md

```diff
@@ -0,0 +1,169 @@
+- `error`
+- `warn` / `warning`
+- `info`
+- `debug`
+- `trace`
+
+#### `policy.revision` (string, overwritten)
+
+If the overall policy has a `revision` field (inserted by Fleet to track policy changes), its value is copied into the input's `policy.revision` field. This allows individual inputs (like Endpoint) to detect policy changes more easily.
+
+
+### Output fields
+
+#### `enabled` (boolean, removed)
+
+If present, this field determines whether the output is active. Defaults to true.
+
+#### `type` (string, required)
+
+The output type. If `use_shipper` is `false`, this value must match one of the entries in the `outputs` field for its inputs' spec files. Otherwise, the `shippers` field for its inputs' spec file must include a shipper type that supports this output. See [Component Specs](component-specs.md) for more details.
+
+#### `use_shipper` (boolean, removed)
+
+If present, this field determines whether this output should be implemented by a Shipper component. Defaults to false.
+
+#### `log_level` (string, removed)
+
+The log level for this component. This field is not passed on to the underlying component; instead, Agent implements log level filtering itself. Possible values:
+- `error`
+- `warn` / `warning`
+- `info`
+- `debug`
+- `trace`
+
+#### `headers` (`map[string]string`)
+
+Agent does not use this field itself, however if the output's `type` is `elasticsearch` then Agent will insert any headers it acquired during Fleet enrollment into this field.
+
+
+### Shipper-specific fields
+
+When components use the shipper, it results in units that don't correspond directly to a configuration entry in the policy. A component that writes to the shipper will be given an output unit that targets the shipper, and a shipper component will be given input units detailing the components that will connect to it.
+
+#### Shipper output fields
+
+The output unit of a component that writes to a shipper is given the following configuration:
+
+- `type` (string): the shipper type
+- `server` (string): the connection address of the shipper (a named socket on Darwin/Linux, a named pipe on Windows)
+- `ssl.certificate_authorities` (string list): a list consisting of one element, which is the certificate authority the shipper will use to verify clients that connect to it. Each shipper instance is assigned its own unique certificate authority on startup.
+- `ssl.certificate` (string): the certificate to present when connecting to the shipper, signed by the CA in `ssl.certificate_authorities`.
+- `ssl.key` (string): the key for `ssl.certificate`
+
+#### Shipper input fields
+
+For each component that writes to a shipper, the shipper will be given an input unit with the following configuration:
+
+- `id` (string): the id of the input unit, which is also the id of the originating component.
+- `type` (string): the shipper type.
```

Comment:
The shipper type -- this is likely redundant, I can't think what it would be used for, but it's what is currently passed. (the input types are in `units[...].config` immediately below, though those also are generally not needed by the shipper.)

Reply:
The shipper type is needed here, so in the case that filebeat becomes the shipper it knows that this input is **a** shipper. Each unit configuration is required because the filebeat should processes that and that is how it should setup the processors.

Reply:
Well, the same Filebeat process will never be both a shipper and a regular input (so far all the code we've written in practice knows it's running as a shipper before checking the input unit policy). But you're right, the processor config is important :-)